### ISSUE TITLE
DBUnit escape pattern on multiple database tests

### ISSUE THREAD

**Issue by muhammadkholidb** [database-rider/database-rider#347]
I have my tests on multiple databases (H2, PostgreSQL, MariaDB), and I want to use an escape pattern specific to the database test currently running. Here is my test configuration:

```java
@DBRider
@DBUnit(caseSensitiveTableNames = true, escapePattern = "\"")
@SpringJUnitConfig(RepositoryTestBase.Config.class)
public abstract class RepositoryTestBase {

    @Configuration
    public static class Config {

    }
}
```

The test classes will extend the abstract class above and be annotated with `@DataSet` for the sample data. On the class above, I use escape pattern `"` which works for H2 and PostgreSQL, while on MariaDB I get SQL syntax error below for table name `path`:

```java
org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '"path"' at line 1
```

What is the best way to make it possible to set escape pattern on multiple database tests? Thank you.

**Comment by rmpestano** [comment]
Hi, not sure we support this right now but I think if we add support for system properties on the dbunit configuration that would work (assuming once you run your tests they all would run against the same db).

So instead instead of declaring the escape pattern at the test level you would have to pass a system property on each run, ex: `mvn test -D dbunit.escapePattern=\"` (need to check the escaping on the command like)

If someone else have other ideas, please share here

**Comment by muhammadkholidb** [comment]
Hi @rmpestano, thanks for the reply. I just know that such an option exists. That's a nice feature of DBRider and can be a solution for my case actually, I  saw the code on `GlobalConfig`. 

But I think it is a good idea to have the escape string detected automatically since the `DatabaseMetaData` has an API for that i.e. `getIdentifierQuoteString()`. I saw the code of `DataSetExecutorImpl` (which is the default executor) makes use of `DatabaseMetaData` to get table names.

And I also saw that the `RESERVED_TABLE_NAMES` value is set to these 3 words: `user`, `password`, `value` when there is no system property set for that. Those 3 words are not enough actually to be used as default values for reserved words, imho. `DatabaseMetaData` has an API `getSQLKeywords()` to get the information which can be used as the default values for reserved words for specific JDBC driver.

My concern is DBRider should have good defaults so that the users can use it with as minimal effort as possible, while still giving other options if they want some customization. Is there any reason not to use `DatabaseMetaData` for those usages above? Let me know your concern.

**Comment by rmpestano** [comment]
Hey @muhammadkholidb, thank you for your feedback.

I think there are no concerns regarding `DatabaseMetaData`, it's not being used mainly due to `ignorance` 🙈. 

So as I understand we could remove this part: https://github.com/database-rider/database-rider/blob/2c6f6b5510ab15513d59e21282b8661a8b39ca9d/rider-core/src/main/java/com/github/database/rider/core/util/TableNameResolver.java#L39-L41

and just rely on dbunit `DatabaseMetaData#getSQLKeywords()` to get the reserved table names. 

Also, we could remove all the code below: https://github.com/database-rider/database-rider/blob/2c6f6b5510ab15513d59e21282b8661a8b39ca9d/rider-core/src/main/java/com/github/database/rider/core/util/TableNameResolver.java#L124-L139

and use  `DatabaseMetaData#getIdentifierQuoteString()`, is that correct?

If you'd like to open a PR I can assist you otherwise I can look later.

Thank you!

**Comment by muhammadkholidb** [comment]
Nice, I'll create a PR for that. Thank you

**Comment by rmpestano** [comment]
The fix for this issue is available on v`1.28.0` which was just released, thank you for the contribution @muhammadkholidb!
